app/Main.hs

```diff
@@ -62,11 +64,20 @@ main = do
 
     runCardanoApplicationWithFeatures Development cardanoFeatures (cardanoApplication nodeLayer)
 
-initializeAllFeatures :: CLIArguments -> CardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
-initializeAllFeatures (CLIArguments logCli nodeCli) cardanoConfiguration cardanoEnvironment = do
-
-    (loggingLayer, loggingFeature) <- createLoggingFeature cardanoEnvironment cardanoConfiguration logCli
-    (nodeLayer   , nodeFeature)    <- createNodeFeature loggingLayer nodeCli cardanoEnvironment cardanoConfiguration
+initializeAllFeatures :: CLIArguments -> PartialCardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
+initializeAllFeatures (CLIArguments logCli nodeCli) partialConfig cardanoEnvironment = do
+    finalConfig <- either (throwIO . ConfigurationError) pure $
+          finaliseCardanoConfiguration $
+          -- Here we perform merging of layers of configuration, but for now, only in a trivial way,
+          -- just for Cardano.Shell.Constants.Types.Genesis.
+          -- We expect this process to become generic at some point.
+          partialConfig { pccCore =
+                          flip fmap (pccCore partialConfig) $
+                          \x -> x { pcoGenesis            = (<>) <$> pcoGenesis            x <*> genesisSpec      nodeCli
```

Comment:
I think that's a sign that we should not define everything inside a `Last`.  You'd have here just `<>`, it should be just `<>`. With this design there are just too many monoids available, do we need them all?

Reply:
Hmm, this particular comment pertains to the code that no longer exists -- that piece of code was removed in `42f4e3b78d1eaf30ff97d37d5cc00f6de5ba05bb`.

Something odd with github?